xbatcher/loaders/keras.py

```diff
@@ -0,0 +1,73 @@
+from typing import Any, Callable, Optional, Tuple
+
+import tensorflow as tf
+import xarray as xr
+
+# Notes:
+# This module includes one Keras dataset, which can be provided to model.fit().
+#  - The CustomTFDataset provides an indexable interface
+# Assumptions made:
+#  - The dataset takes pre-configured X/y xbatcher generators (may not always want two generators in a dataset)
+
+
+class CustomTFDataset(tf.keras.utils.Sequence):
+    def __init__(
+        self,
+        X_generator,
+        y_generator,
+        *,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+        dim: str = 'new_dim',
+    ) -> None:
+        '''
+        Keras Dataset adapter for Xbatcher
+
+        Parameters
+        ----------
+        X_generator : xbatcher.BatchGenerator
+        y_generator : xbatcher.BatchGenerator
+        transform : callable, optional
+            A function/transform that takes in an array and returns a transformed version.
+        target_transform : callable, optional
+            A function/transform that takes in the target and transforms it.
+        dim : str, 'new_dim'
```

Comment:
You have just taught me that default values can go after the type definition 🙇‍♂️